src/java/org/apache/cassandra/db/commitlog/SimpleCachedBufferPool.java

```diff
@@ -79,12 +88,12 @@ public ByteBuffer createBuffer(BufferType bufferType)
 
     public ByteBuffer getThreadLocalReusableBuffer(int size)
     {
-        ByteBuffer result = reusableBufferHolder.get();
-        if (result.capacity() < size || BufferType.typeOf(result) != preferredReusableBufferType)
+        ByteBuffer result = reusableBBHolder.get(preferredReusableBufferType).get();
```

Comment:
Nit: Instead of `preferredReusableBufferType` we can store the result of `reusableBBHolder.get(preferredReusableBufferType)` on `setPreferredReusableBufferType` calls to avoid the `EnumMap` lookup.

Reply:
Not so nit imo. Getting the buffer is sure more prone to being part of a hot path than changing the type. So fiddly but I gave it a go.